openwisp_controller/config/base/template.py

```diff
@@ -152,6 +152,8 @@ def clean(self, *args, **kwargs):
         if self.type == 'vpn' and not self.config:
             self.config = self.vpn.auto_client(auto_cert=self.auto_cert)
         super().clean(*args, **kwargs)
+        if not self.config:
+            raise ValidationError(_('Configuration is required'))
```

Comment:
It should be enough to remove this: https://github.com/openwisp/openwisp-controller/blob/master/openwisp_controller/config/base/template.py#L193

I wonder why I added it. Doesn't make much sense to me nowadays.

Reply:
Ok. Let me fix this

Reply:
@nemesisdesign on a second thought, this will not be the best thing to do because there are instances where the template's config is gotten from other objects (for example: a VPN object [using this line](https://github.com/openwisp/openwisp-controller/blob/master/openwisp_controller/config/base/template.py#L153). Making `config` field required this manner will cause `clean_field` validation to fail before `clean method` is called as explained [in django docs](https://docs.djangoproject.com/en/3.1/ref/models/instances/#validating-objects)

I suggest to leave the solution as this but improve abit on the UI.